libminifi/include/utils/StringView.h

```diff
@@ -0,0 +1,151 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+
+#include <string>
+#include <iterator>
+#include <algorithm>
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace utils {
+
+class StringView {
```

Comment:
I would opt for `string_view` and a using declaration for `std::string_view` instead of the implementation when it's available. That would also require at least partial compatibility with `std::string_view`.

Reply:
as we plan on upgrading to `c++17` I am against introducing anything "standard-conformant-aiming", moreover it also provides partial compatibility in the form of "iterability", but hopefully as soon as we have `std::string_view`, this class becomes obsolete

Reply:
although the utility of this class itself might be questioned (I just couldn't stomach copying one more `std::string` when a non-copying mechanism would suffice) should we put out this class of its (short) misery?

Reply:
I'm fine with either removing or renaming this. Why are you against "standard-conformant-aiming"? The purpose would be to ease the upgrade by making the standard version a drop-in replacement.

Reply:
as a general rule of thumb, I favour granularity, I believe we should promise as little as possible (because promises have to be kept and that takes effort), and features should be added on a as-needed basis, for example the lack of `operator[]` in this case makes this implementation non-conformant, but that is inconsequential

(now on the other hand the added methods `startsWith` and `endsWith` (coming in `c++20`) could have pose a problem when "upgrading" to `std::string_view`, this is a shortcoming of the language offering no extension methods, that is why I believe the best course of action is to centralize the introduction of stl parts, so we may use whatever implementation we want, so no `std::vector`, but `minifi::std::vector`)

Reply:
`StringView` has been eliminated